### ISSUE TITLE
Divergent loss with SimCLR

### ISSUE THREAD

**Issue by antho214** [lightly-ai/lightly#1633]
Sometimes, when training using the SimCLR method I get some divergent loss function (see attached screenshot). I wonder if anyone has ever experienced this kind of issue when training with SimCLR. This has happened to me on different occasions with ResNet-18/50 model.
I don't think that this is an issue with the code, but if anyone has ever seen this kind of problem I would be grateful for your input.

![Screenshot 2024-08-15 at 07 45 03](https://github.com/user-attachments/assets/5731047b-9943-4c27-8d9e-91efd86d6493)

Here's some information about the training hyper-parameters:
- multi-gpu training with `batch_size` of 256 per GPU
- `batch_size`: 1024
- `criterion`: `NTXentLoss` with temperature of 0.1 and `gather_distributed=True`
- `optimizer`: LARS with a base learning rate of 0.3 and default parameters from [here](https://github.com/lightly-ai/lightly/blob/222d97b5d3f8085604abe1bce612a4ffd573a312/benchmarks/imagenet/resnet50/simclr.py)
- `scheduler`: `CosineWarmupScheduler` with 10k warmup steps

**Comment by guarin** [comment]
This looks interesting, I haven't encountered this before. What type of data are you using? And do you use sync batchnorm?

**Comment by IgorSusmelj** [comment]
I could imagine that you face some numerical instabilities

- do you use fp16 or bf16? (in case you use fp16 try switching to bf16 as it might be more stable)
- you could try to plot the histogram of the weights in tensorboard to see if some values are getting very large --> if yes, you could try use additional gradient clipping, weight clipping, stronger weight decay

**Comment by antho214** [comment]
Thank you for taking the time to answer.

- I do not use the 16-bit precision when training. I am using the default value of the `Trainer` which is `32-true` according to the documentation.
- I will investigate whether the weights are getting large using tensorboard.
- I am tracking the augmentations over time and they do look different.

Again, thank you for the feedback. I will update if I find a fix/solution.
